root/static/scripts/common/components/CDTocReleaseListRow.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import ReleaseCatnoList from '../../../../components/ReleaseCatnoList.js';
+import ReleaseLabelList from '../../../../components/ReleaseLabelList.js';
+import {SanitizedCatalystContext} from '../../../../context.mjs';
+import MediumTracklist from '../../../../medium/MediumTracklist.js';
+import loopParity from '../../../../utility/loopParity.js';
+import type {ReleaseWithMediumsAndReleaseGroupT}
+  from '../../relationship-editor/types.js';
+import bracketed from '../utility/bracketed.js';
+import formatBarcode from '../utility/formatBarcode.js';
+import mediumFormatName from '../utility/mediumFormatName.js';
+
+import ArtistCreditLink from './ArtistCreditLink.js';
+import EntityLink from './EntityLink.js';
+import ReleaseEvents from './ReleaseEvents.js';
+import TaggerIcon from './TaggerIcon.js';
+
+type Props = {
+  +associatedMedium?: number,
+  +cdTocTrackCount: number,
+  +countInReleaseGroup: number,
+  +release: ReleaseWithMediumsAndReleaseGroupT,
+  +showArtists?: boolean,
+  +wasMbidSearch?: boolean,
+};
+
+const CDTocReleaseListRow = ({
+  associatedMedium,
+  cdTocTrackCount,
+  countInReleaseGroup,
+  release,
+  showArtists = false,
+  wasMbidSearch = false,
+}: Props): React.Element<typeof React.Fragment> => {
+  const $c = React.useContext(SanitizedCatalystContext);
+
+  const [hidden, setHidden] = React.useState<boolean>(true);
```

Comment:
On the TT page it looks like you were able to expand/collapse each medium individually, whereas this controls all of them? (Even though there are still separate links for each medium, they all expand all mediums, which seems unintended.)

Reply:
Hmm. I cannot set the state for each medium inside `attachableMediums.map`, right? - would you suggest having some sort of object with id and a boolean for hidden, or having a different component for the medium display that gets called inside `attachableMediums`, or something else entirely?

Reply:
Moved the code to display each medium to a new component with its own state which means these actually work separately from each other now.